src/applications/personalization/profile-2/components/Profile2Router.jsx

```diff
@@ -63,6 +67,13 @@ class Profile2Router extends Component {
     ) {
       this.props.fetchPaymentInformation();
     }
+
+    if (
+      this.props.shouldShowMilitaryInformation &&
+      !prevProps.shouldShowMilitaryInformation
```

Comment:
Yeah - same here! I'm pretty sure we're gonna convert the whole thing into a functional component here soon! I'd spend some time doing it now, but am reluctant to include the change into this PR (and changing it might also require quite a few changes to the existing unit tests, don't you think?)

Reply:
Oh totally, no rush to do the conversion!